dashboard/src/components/loading/AnimatedDashboardLogo.tsx

```diff
@@ -0,0 +1,110 @@
+'use client';
+
+import React, { memo, useId } from 'react';
+
+interface AnimatedDashboardLogoProps {
+  size?: number;
+  className?: string;
+}
+
+export const AnimatedDashboardLogo = memo(function AnimatedDashboardLogo({
+  size = 80,
+  className = '',
+}: AnimatedDashboardLogoProps) {
+  const id = useId();
+  const maskId = `logoMask-${id}`;
+
+  return (
+    <div
+      className={`relative ${className}`}
+      style={{
+        width: size,
+        height: size,
+        contain: 'layout style paint',
+      }}
+    >
+      <svg width={size} height={size} viewBox='0 0 1071 1069' fill='none' xmlns='http://www.w3.org/2000/svg'>
+        <defs>
+          <mask id={maskId}>
+            <rect x='0' y='0' width='375' height='1069' fill='white' className='animate-fill-up-1' />
+            <rect x='375' y='0' width='375' height='1069' fill='white' className='animate-fill-up-2' />
+            <rect x='750' y='0' width='321' height='1069' fill='white' className='animate-fill-up-3' />
+          </mask>
+        </defs>
```

Comment:
```suggestion
        <defs>
          <mask id={maskId}>
            <rect
              x={0}
              y={0}
              width={(375 / 1071) * size}
              height={size}
              fill="white"
              className="animate-fill-up-1"
            />
            <rect
              x={(375 / 1071) * size}
              y={0}
              width={(375 / 1071) * size}
              height={size}
              fill="white"
              className="animate-fill-up-2"
            />
            <rect
              x={(750 / 1071) * size}
              y={0}
              width={(321 / 1071) * size}
              height={size}
              fill="white"
              className="animate-fill-up-3"
            />
          </mask>
        </defs>
```

Scale mask rectangles based on component size instead of using fixed pixel dimensions to avoid same problem I mention in the other comment.